Docs/add uploader docs (#1382)
reimagined/resolve: docs/advanced-techniques.md

<!-- prettier-ignore-end -->

> A merged module's code is included into the resulting application's bundles and behaves as a part of the application.

Reviewer: What do you mean "and behaves"?

Author: The code behaves

Reviewer: How can code 'behave'?

Author: Changed to "is interpreted as a part of the application"

> A merged module's code is included into the resulting application's bundles and is interpreted as a part of the application.

Reviewer: ```suggestion
A merged module's code is included in the resulting application's bundles.
```

> For an example on how to use modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application uses the authentication module and the comments module that reSolve provides.

Reviewer: ```suggestion
For an example on how to use modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application uses the authentication and comments modules from reSolve.
```

## Upload Files

The **resolve-module-upload** module implements the file upload functionality. You can enable this module as shown below:

##### run.js:
```js
const moduleUploader = resolveModuleUploader({ jwtSecret })
...
const baseConfig = merge(
  defaultResolveConfig,
  appConfig,
  moduleAuth,
  moduleUploader
)
```

The **resolve-module-upload** module adds the following API endpoints to an application:

- `/api/uploader/getFormUpload` - Returns an upload path to use in HTTP forms.
- `/api/uploader/getUploadUrl` - Returns a path used to upload files.
- `/api/uploader/getToken` - Takes user credentials and returns the user's authorization token.

> The [cli-uploader](https://github.com/reimagined/resolve/tree/master/examples/cli-uploader) example application demonstrates how to design a file uploader utility and handle file uploads on the server side.

Reviewer: ```suggestion
The [cli-uploader](https://github.com/reimagined/resolve/tree/master/examples/cli-uploader) example application demonstrates how to design a file uploader utility and handle file uploads on the server.
```

> A merged module's code is treated the same as the application's code. The resulting application's bundles include the module code and configurations as if they were always a part of the application.

Reviewer: This is a strange paragraph. "is treated the same"? " as if they were always a part"?

Author: Rewrote the paragraph like so:
"A merged module's code is included into the resulting application's bundles and behaves as a part of the application."

> For an example on using modules, see the [Hacker News](https://github.com/reimagined/resolve/tree/master/examples/hacker-news) sample application. This application makes use of the authentication module and the comments module provided by reSolve.

Reviewer: "on using modules" Do you mean 'on how to use modules' or 'how modules are used'?

Reviewer: "makes use of the" So you mean it 'uses'?

Reviewer: "the comments module provided by reSolve" Does this need to be in passive voice?

Author: > "on using modules" Do you mean 'on how to use modules' or 'how modules are used'?

Changed to "on how to use modules"

Author: > "makes use of the" So you mean it 'uses'?

Yes, changed to "uses"

Author: Changed to "the comments module that reSolve provides"

## Upload Files

The **resolve-module-upload** module implements the file upload functionality. You can enable this module as shown below:

##### run.js:
```js
const moduleUploader = resolveModuleUploader({ jwtSecret })
...
const baseConfig = merge(
  defaultResolveConfig,
  appConfig,
  moduleAuth,
  moduleUploader
)
```

The **resolve-module-upload** module adds the following API endpoints to an application:

> - `/api/uploader/getFormUpload` - Returns an upload path to use in HTTP forms.

Reviewer: "an upload path to use in HTTP forms" Clarify

Author: For example, if you add an "Upload File" form to your web application's page, you need to specify where to upload files when this form is submitted (upload path). The described endpoint returns this path.

Reviewer: ```suggestion
- `/api/uploader/getFormUpload` - Returns the upload path used in HTTP forms.
```

- `/api/uploader/getUploadUrl` - Returns a path used to upload files.

> - `/api/uploader/getToken` - Given credentials returns the user's authorization token.

Reviewer: "Given credentials" Which 'given' credentials?

Author: Changed to "Takes user credentials and returns the user's authorization token."
